tools/integration/src/index.ts

```diff
@@ -758,123 +817,359 @@ async function handleExport(argv: any) {
+        const allEvents = await getEventList(server, token, axiosInstance);
+        let totalEventProcessed = 0;
+
+        for (const inc of parsedIncludes.filter(inc => inc.type === "event" || inc.type === "all")) {
+            const matches = inc.conditions.filter(c => c.startsWith("id="));
+
+            let filtered;
+            if (matches.length && inc.explicitlyTyped) {
+                filtered = matches.map(idCond => {
+                    const id = idCond.split("=")[1]?.replace(/^"|"$/g, '');
+                    return { id };
+                });
+            } else if (matches.length) {
+                filtered = matches.map(idCond => {
+                    const id = idCond.split("=")[1]?.replace(/^"|"$/g, '');
+                    return allEvents.find(e => e.id === id);
+                }).filter(Boolean);
+            } else {
+                filtered = filterEventsBy(allEvents, inc.conditions);
+            }
 
-            const response = await axiosInstance.get(url, {
-                headers: {
-                    'Content-Type': 'application/json',
-                    'Authorization': `apiToken ${token}`
+            if (filtered.length === 0) {
+                const logFn = inc.explicitlyTyped ? logger.error : logger.debug;
+                logFn(`No custom event(s) found matching: ${inc.conditions.join(', ')}`);
+                continue;
+            }
+		const enriched = filtered.map(item => ({
+                ...item,
+                name: item.name ?? `custom-event-${item.id}`
+            }));
+            const sanitized = sanitizeTitles(filtered, "custom-event");
+            for (const evt of sanitized) {
+                const event = await exportEvent(server, token, evt.id, axiosInstance);
+                if (event) {
+                    saveEvent(eventsPath, evt.id, evt.title, event);
+                    totalEventProcessed++;
+                    wasEventFound = true;
+                } else {
+                    const logFn = inc.explicitlyTyped ? logger.error : logger.debug;
+                    logFn(`Event with id=${evt.id} not found or failed to export.`);
                 }
-            });
-            logger.info(`Successfully got the dashboard list: ${response.status}`);
-            if (logger.isDebugEnabled()) {
-                logger.debug(`Response data: \n${JSON.stringify(response.data)}`);
             }
+        }
+
+        logger.info(`Total custom event(s) processed: ${totalEventProcessed}`);
+    }
+
+    // Final info
+    const isOnlyDashboard = parsedIncludes.every(inc => inc.type === "dashboard");
+    const isOnlyEvent = parsedIncludes.every(inc => inc.type === "event");
+    const isMixedOrUnspecified = parsedIncludes.some(inc => inc.type === "all") || (!isOnlyDashboard && !isOnlyEvent);
+
+    if (!wasDashboardFound && !wasEventFound && isMixedOrUnspecified) {
+        logger.error("No custom elements were found or exported.");
+    }
```

Comment:
The code logic here is tricky. Shall we just simply count the processed dashboards and events, and if there's nothing processed, we output no custom elements exported. We should not rely on the presence of type param in the `--include` clause, because it could be possible that no element is processed even the type is specified, e.g.: no match found for this type.